utils/deployment-util.js

```diff
@@ -131,78 +131,105 @@ const createExtensions = async ({ dao, factories, options }) => {
+        tx = await factory.create();
+      }
 
-      extensionAddress = factoryEvent.args[1];
-    } else {
-      const { logs } = tx;
-      const factoryLog = logs.find((l) => l.address === factory.address);
-      if (!factoryLog) {
-        throw new Error("no event emitted by the factory");
+      /**
+       * The tx event is the safest way to read the new extension address.
+       * 1. Find the event that contains the `args` field which indicates the factory event
+       * 2. Take the data at index 1 which represents the new extension address
+       */
+      let extensionAddress;
+      if (tx.wait) {
+        info(
+          `waiting for transaction ${tx.hash} to be mined... nonce: ${tx.nonce}`
+        );
+        const res = await tx.wait();
+        const factoryEvent = res.events.find(
+          (e) => e.address === factory.address
+        );
+        if (!factoryEvent) throw new Error("Missing factory event.");
+
+        extensionAddress = factoryEvent.args[1];
+      } else {
+        const { logs } = tx;
+        const factoryLog = logs.find((l) => l.address === factory.address);
+        if (!factoryLog) {
+          throw new Error("no event emitted by the factory");
+        }
+        extensionAddress = factoryLog.args[1];
       }
-      extensionAddress = factoryLog.args[1];
-    }
-    const extensionInterface = options[extensionConfigs.name];
-    if (!extensionInterface)
-      throw new Error(
-        `Extension contract not found for ${extensionConfigs.name}`
+
+      const extensionInterface = options[extensionConfigs.name];
+      if (!extensionInterface)
+        throw new Error(
+          `Extension contract not found for ${extensionConfigs.name}`
+        );
+
+      const newExtension = embedConfigs(
+        await options.attachFunction(extensionInterface, extensionAddress),
+        extensionInterface.contractName,
+        options.contractConfigs
       );
 
-    const newExtension = embedConfigs(
-      await options.attachFunction(extensionInterface, extensionAddress),
-      extensionInterface.contractName,
-      options.contractConfigs
-    );
+      if (!newExtension || !newExtension.configs)
+        throw new Error(
+          `Unable to embed extension configs for ${extensionConfigs.name}`
+        );
 
-    if (!newExtension || !newExtension.configs)
-      throw new Error(
-        `Unable to embed extension configs for ${extensionConfigs.name}`
+      await waitTx(
+        dao["addExtension(bytes32,address)"](extensionId, newExtension.address)
       );
 
-    await waitTx(
-      dao["addExtension(bytes32,address)"](
-        sha3(newExtension.configs.id),
-        newExtension.address
-      )
-    );
+      info(`
+      Extension enabled '${newExtension.configs.name}'
+      -------------------------------------------------
+      contract address: ${newExtension.address}
+      creator address:  ${options.owner}
+      identity address: ${factory.identity}
+      `);
```

Comment:
I am wondering if we should move this log instruction to line 230. I think we should see in the logs if the extension was enabled even if it was resumed.